GVFS/GVFS.Common/Git/GitSsl.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using GVFS.Common.Tracing;
+
+namespace GVFS.Common.Git
+{
+    public class GitSsl
+    {
+        private readonly string certificatePathOrSubjectCommonName;
+        private readonly bool isCertificatePasswordProtected;
+
+        public GitSsl()
+        {
+            this.certificatePathOrSubjectCommonName = null;
+            this.isCertificatePasswordProtected = false;
+            this.ShouldVerify = true;
+        }
+
+        public GitSsl(IDictionary<string, GitConfigSetting> configSettings) : this()
+        {
+            if (configSettings != null)
+            {
+                if (configSettings.TryGetValue(GitConfigSetting.HttpSslCert, out GitConfigSetting sslCerts))
+                {
+                    this.certificatePathOrSubjectCommonName = sslCerts.Values.Single();

```

Comment:
`Single` can throw.  Do we have a guarantee at this point that there is exactly one item in `sslCerts.Values`?  If not, do we need to handle the exception (or check that there's exactly one item first)?

(Same comment applies elsewhere in this method where we're using `Single`)

Reply:
Currently, if you have multiple git settings, like
```
http.sslcert = cert1
http.sslcert = cert2
```

Git uses the last one. With that in mind, I'll replace Single() with Last(), although with current git behavior, it should only output one line per config key anyway.